src/accelerate/state.py

```diff
@@ -724,13 +724,13 @@ def _prepare_backend(
             elif is_musa_available():
                 backend = "mccl"
                 distributed_type = DistributedType.MULTI_MUSA
+            elif is_npu_available():
+                backend = "hccl"
+                distributed_type = DistributedType.MULTI_NPU
```

Comment:
agreed with @BenjaminBossan, let's add a comment here and during `torch.device("npu")` that clarifies (in the code) that these must be done before the cuda check